pandora-common/src/networking/directory_client.ts

```diff
@@ -101,6 +101,40 @@ export const ACCOUNT_SETTINGS_DEFAULT = Object.freeze<IDirectoryAccountSettings>
 	interfaceChatroomOfflineCharacterFilter: 'ghost',
 });
 
+export const DirectoryAccountSettingsLimitedSchema = z.object({
+	displayName: DisplayNameSchema,
+});
+export type IDirectoryAccountSettingsLimited = z.infer<typeof DirectoryAccountSettingsLimitedSchema>;
+
+type DirectoryAccountSettingsLimitedStoredSchemaAcc = {
+	[K in keyof IDirectoryAccountSettingsLimited]: ZodObject<{
+		value: typeof DirectoryAccountSettingsLimitedSchema.shape[K];
+		nextAllowedChange: ZodNumber;
+	}>;
+};
+export const DirectoryAccountSettingsLimitedStoredSchema = z.object(
+	KnownObject.entries(DirectoryAccountSettingsLimitedSchema.shape).reduce(
+		(acc, [key, value]) => {
+			acc[key] = z.object({
+				value,
+				nextAllowedChange: z.number(),
+			});
+			return acc;
+		},
+		// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+		{} as DirectoryAccountSettingsLimitedStoredSchemaAcc,
+	),
+);
+export type IDirectoryAccountSettingsLimitedStored = z.infer<typeof DirectoryAccountSettingsLimitedStoredSchema>;
+
+export const ACCOUNT_SETTINGS_LIMITED_STORED_DEFAULT = Object.freeze<IDirectoryAccountSettingsLimitedStored>({
+	displayName: { value: '', nextAllowedChange: 0 },
```

Comment:
This default value isn't actually valid - the `DisplayNameSchema` specifies minimum length of 3.